tests/queries/0_stateless/03443_continue_insert_with_progress_headers_on_client_close.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+# Tags: no-parallel, no-fasttest
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+${CLICKHOUSE_CURL} -sS "${CLICKHOUSE_URL}" -d 'CREATE TABLE numbers (x int) ENGINE = Memory();'
+
+# Query takes 10 seconds and sends back progress headers, we close the connection after 5 seconds and expect the query to continue
+${CLICKHOUSE_CURL} --max-time 5 -sS "${CLICKHOUSE_URL}&query_id=continue_insert_on_network_issue&max_threads=1&send_progress_in_http_headers=1" --data-binary $'INSERT INTO numbers SELECT sleepEachRow(0.1) + number as x from system.numbers limit 100 settings function_sleep_max_microseconds_per_block = 0, send_progress_in_http_headers = 1, insert_deduplicate = 0, max_block_size = 10;'  2>&1 | grep -cF 'curl: (28)'
```

Comment:
Let's also check that it writes a "query finish" entry to query_log
(I guess it may write an "exception while processing entry" because it will probably fail to send query status to the client in the very end, although it will not roll back the result of INSERT SELECT)